utils/frame/remote-externalities/src/lib.rs

```diff
@@ -351,14 +351,57 @@ where
 		prefix: StorageKey,
 		at: B::Hash,
 	) -> Result<Vec<StorageKey>, &'static str> {
-		let mut last_key: Option<StorageKey> = None;
+		const NUM_PARTS: usize = 4; // number of parts to divide the prefix into
+
+		let mut tasks = Vec::with_capacity(NUM_PARTS);
+
+		// Divide the prefix into N parts
+		let mut start_key = prefix.clone();
+		let mut end_key = prefix.clone();
 		let mut all_keys: Vec<StorageKey> = vec![];
-		let keys = loop {
+
+		for _ in 0..NUM_PARTS {
+			end_key.0[0] = start_key.0[0].wrapping_add(1);
```

Comment:
This line panics when run against Polkadot state:

```
$ cargo run --features=try-runtime try-runtime --runtime ../polkadot/target/release/wbuild/polkadot-runtime/polkadot_runtime.wasm on-runtime-upgrade live --uri ws://localhost:9944
...
Thread 'main' panicked at 'index out of bounds: the len is 0 but the index is 0', /Users/liamaharon/grimoire/substrate/utils/frame/remote-externalities/src/lib.rs:364
```

How're you testing it?

Reply:
Earlier I had only build the node with --features try-runtime flag in release mode. Thanks for the comments, I've updated the implementation.